shared/AppInsightsCommon/src/Util.ts

```diff
@@ -355,7 +355,7 @@ export const CorrelationIdHelper: ICorrelationIdHelper = {
             requestHost = (urlParseFullHost(requestUrl, true) || "").toLowerCase();
         }
 
-        if ((!config || !config.enableCorsCorrelation) && requestHost !== currentHost) {
+        if ((!config || !config.enableCorsCorrelation) && (requestHost && requestHost !== currentHost)) {
```

Comment:
ONLY check if the requestHost equals currentHost when requestHost exists. For the case that requestHost not exist (empty string when host not exist) skip the check. Otherwise it will returns false right away for the cases that url without host.